### ISSUE TITLE
CA1069 throws AD0001

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn-analyzers#3730]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v3.0.0 (Latest)

#### Diagnostic ID

Example: [CA1069](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1069)

#### Repro steps

I had this issue while doing some refactoring in an enum I was creating by removing the value associated to the field. I couldn't reproduce the issue but thought it might still be worth reporting it.

```
Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines.EnumShouldNotHaveDuplicatedValues' threw an exception of type 'System.ArgumentNullException' with message 'Value cannot be null.
Parameter name: key'
```

**Comment by mavasani** [comment]
@Evangelink wondering if value can be null below: https://github.com/dotnet/roslyn-analyzers/blob/f15404b312295e7cee16fb40b4c8d3f91f10f087/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/EnumShouldNotHaveDuplicatedValues.cs#L114

**Comment by Evangelink** [comment]
I was actually wondering about this line:
https://github.com/dotnet/roslyn-analyzers/blob/f15404b312295e7cee16fb40b4c8d3f91f10f087/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/EnumShouldNotHaveDuplicatedValues.cs#L95

There is a check done a couple of lines before but as I was changing the code it's possible that the value is now `null`.

On your example, the code is for cases where the value is implicit so I would be less worried.

Except for playing defensively in both cases, I don't see how I could create a test-case.

**Comment by mavasani** [comment]
@Evangelink Yes, let us make the code defensive here. I think we should fix this issue before release 3.3.0-beta1.
